### PR TITLE
fix(compiler): adding the inert property to the "SCHEMA" array

### LINKED ISSUE #51879: Template binding does not recognize the 'inert' property.
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

The 'inert' property under HTMLElement can be accessed correctly in TypeScript, but it cannot be used with template binding.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/i7jj8y?file=src%2Fapp%2Fapp.component.html

### Please provide the exception or error you saw

```true
error NG8002: Can't bind to 'inert' since it isn't a known property of 'a'.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 16.2.3
Node: 16.20.0
Package Manager: npm 9.4.2
OS: linux x64

Angular: 16.2.6
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1602.3
@angular-devkit/build-angular   16.2.3
@angular-devkit/core            16.2.3
@angular-devkit/schematics      16.2.3
@angular/cli                    16.2.3
@schematics/angular             16.2.3
rxjs                            7.8.1
typescript                      4.9.5
zone.js                         0.13.3
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by physicx237**
This change allows template binding "inert" attribute with the following syntax: [inert]="isInert"

Fixes #51879

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #51879

## What is the new behavior?
Possibility of template binding of the inert attribute with the following syntax: [inert]="isInert"

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by JeanMeche** [comment]
Cf the file header

> //                       DO NOT EDIT THIS DOM SCHEMA WITHOUT A SECURITY REVIEW!
> //
> // Newly added properties must be security reviewed and assigned an appropriate SecurityContext in
> // dom_security_schema.ts. Reach out to mprobst & rjamet for details.

I'm marking it as blocked until this is done.

File: packages/compiler/src/schema/dom_element_schema_registry.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ const SCHEMA: string[] = [
...
...able,dir,!draggable,enterKeyHint,!hidden,!inert,innerText,inputMode,lang,nonce,*abort,*animationend,*animationiteration,*animationstart,*auxclick,*beforexrselect,*blur,*cancel,*canplay,*canplaythrough,*change,*click,*close,*contextmenu,*copy,*cuechange,*cut,*dblclick,*drag,*dragend,*dragenter,*dragleave,*dragover,*dragstart,*drop,*durationchange,*emptied,*ended,*error,*focus,*formdata,*gotpointercapture,*input,*invalid,*keydown,*keypress,*keyup,*load,*loadeddata,*loadedmetadata,*loadstart,*lostpointercapture,*mousedown,*mouseenter,*mouseleave,*mousemove,*mouseout,*mouseover,*mouseup,*mousewheel,*paste,*pause,*play,*playing,*pointercancel,*pointerdown,*pointerenter,*pointerleave,*pointermove,*pointerout,*pointerover,*pointerrawupdate,*pointerup,*progress,*ratechange,*reset,*resize,*scroll,*securitypolicyviolation,*seeked,*seeking,*select,*selectionchange,*selectstart,*slotchange,*stalled,*submit,*suspend,*timeupdate,*toggle,*transitioncancel,*transitionend,*transitionrun,*transitionstart,*volumechange,*waiting,*webkitanimationend,*webkitanimationiteration,*webkitanimationstart,*webkittransitionend,*wheel,outerText,!spellcheck,%style,#tabIndex,title,!translate,virtualKeyboardPolicy',
```
Comment (JoostK) [inline packages/compiler/src/schema/dom_element_schema_registry.ts:81]: This code is generated using

https://github.com/angular/angular/blob/c07805612f77ff59104470a40b86bdd28a93bc61/packages/compiler/test/schema/dom_element_schema_registry_spec.ts#L186-L194

by uncommenting that console.log, changing the `it` to `fit` and running:

```
yarn bazel test packages/compiler/test:test_web_chromium --test_output=streamed
```

or possibly

```
yarn bazel run packages/compiler/test:test_web_debug
```

to run it using your local (Chrome) browser, which may be newer than Chrome 114 as used in the tests.

Comment (physicx237) [reply packages/compiler/src/schema/dom_element_schema_registry.ts:81]: Yes, inert is the only addition.

File: packages/compiler/src/schema/dom_security_schema.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export function SECURITY_SCHEMA(): {[k: string]: SecurityContext} {
       'iframe|srcdoc',
       '*|innerHTML',
       '*|outerHTML',
+      '*|inert',
```
Comment (JoostK) [inline packages/compiler/src/schema/dom_security_schema.ts:34]: I don't think this is needed; `inert` is not a property that needs an HTML security context.

**Comment by physicx237** [comment]
JeanMeche said:
"See file header

// DO NOT EDIT THIS HOUSE DIAGRAM WITHOUT A SECURITY CHECK!
//
// Newly added properties must be security checked and match the corresponding SecurityContext in
// dom_security_schema.ts. More information on the mprost & rjamet website.

I mark it as blocked until this is done."

Should I leave the "inert" attribute in the security context or is it not needed?

**Comment by JoostK** [comment]
The intent of that comment is to make people aware of _potential_ security implications of the schema definitions. For example, Angular must sanitize data when binding to the `innerHTML` property, but not when binding to the `inert` property since that will only be interpreted as boolean value, _not_ rendered as HTML.

Hence, changes to the schema require a security review to avoid mistakenly introducing a vulnerability in the framework. For `inert` though, I cannot think of a reason why it would be security sensitive.

**Comment by physicx237** [comment]
> Cf the file header
> 
> > //                       DO NOT EDIT THIS DOM SCHEMA WITHOUT A SECURITY REVIEW!
> > //
> > // Newly added properties must be security reviewed and assigned an appropriate SecurityContext in
> > // dom_security_schema.ts. Reach out to mprobst & rjamet for details.
> 
> I'm marking it as blocked until this is done. 

How to reach out to rjamet & mbprost to details?

**Comment by jelbourn** [comment]
Ah, that comment is out of date (mbprost & rjamet aren't involved with Angular any longer). @bjarkler I believe `inert` should be completely safe here as a boolean attribute, but I'll add you for FYI anyway since it's part of the process for changing the schema. 

LGTM insofar as adding `inert` in general, just need to address the comments from @JoostK

File: packages/compiler/src/schema/dom_security_schema.ts (hunk 2 above)
Comment (bjarkler) [reply packages/compiler/src/schema/dom_security_schema.ts:34]: +1, this should be removed.

**Comment by bjarkler** [comment]
LGTM, thanks!

**Comment by physicx237** [comment]
Thanks for your feedback!

**Comment by JoostK** [comment]
Could you please squash the commits? The revert won't be auto squashed otherwise

**Comment by physicx237** [comment]
Yes, sure

**Comment by JoostK** [comment]
Looks like something went wrong, there's now five commits.

**Review by JoostK — APPROVED** [review]
Thanks for the updates!

**Comment by physicx237** [comment]
And thank you!

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: fw-security

**Comment by physicx237** [comment]
Hello @alxhub, @pkozlowski-opensource and @josephperrott! Can you review my code?

**Comment by physicx237** [comment]
Hello @alxhub, @pkozlowski-opensource and @josephperrott! Can you review my code?

**Review by AndrewKushnir — APPROVED** [review]

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: fw-security

**Comment by alxhub** [comment]
This PR was merged into the repository by commit dba3e0b5aa73d49a6caad4ad474e0212486fe98a.

**Pull request closed without merging**